### ISSUE TITLE
ember electron:publish

### ISSUE THREAD

**Issue by basz** [adopted-ember-addons/ember-electron#637]
I do not believe `ember electron:publish` is a supported command at the moment. 

To publish I need to do...

```
cd electron-app
NUCLEUS_TOKEN=xxxxxx npx electron-forge publish
```
Is this currently the recommended manner?

Is such a command planned as part of ember-electron?

**Comment by rwwagner90** [comment]
I'm unfamiliar with `publish`. I use `ember electron:make` then upload my assets.

**Comment by bendemboski** [comment]
Publish is not currently a supported command. After a brief look, I think the main thing that would need to be figured out would be supporting incremental builds, i.e. building and/or packaging and/or making, and then publishing based off of those already-built assets (like we do for the package and make commands).

The publish command has support for everything we need, so implementing support for publish would probably involve using the a copy of our make command and task as a starting point, figuring out the right `ember-cli` command options to expose in order to support the various scenarios of publishing different stages of pre-built assets (or building everything), and adapting the make command and task implementations accordingly.

PRs welcome!

**Comment by basz** [comment]
Hmm. It seems you cannot run api.publish without it also doing the make process unless you provide makeResults to the api.publish. 

So would I be correct if we follow this logic.

We add a publish *task* that publishes with the makeResults so making is skipped.

we also add a flag to the make *task* to be able to run a publish *command* (ember electron:make --publish=true (defaults to false))

we can then run our make *command* as is but at the end do a publish where we provide the makeResult to the publish command.

(a bit later) I have tested passing and makeResults to api.publish his seems to work! 

`ember electron:publish` would then not exist but `ember electron:make --publish` would

**Comment by bendemboski** [comment]
Fixed by #811
